docs/development.md

```diff
@@ -344,3 +344,145 @@ respectively.
 ```bash
 xtask --logs run-cargo-fuzz --crate-name=loader --target-name=wasm_invoke -- -max_total_time=20
 ```
+
+## Retrieving a previously-built binary and its provenance
+
+Every time a pull-request is merged to `main` the
+`oak_functions_freestanding_bin` binary is built and uploaded to `ent`.
+Similarly, a
+[signed SLSA3+ provenance is generated](https://github.com/slsa-framework/slsa-github-generator/blob/04f1fe0c7b7902c9a95f4c7eef2dc04cf0f8e6a7/internal/builders/generic/README.md)
+for it and uploaded to GitHub. The signed provenance is also uploaded to the
+public instance of Rekor hosted by Sigstore at https://rekor.sigstore.dev, and
+can be downloaded from it.
+
+This section describes how the binary and its provenance can be downloaded.
+
+Once a pull-request is merged to `main` and all CI steps are executed on branch
+`main`, an automated comment is added to the pull-request containing the hash of
+the `oak_functions_freestanding_bin` binary:
+
+```bash
+d059c38cea82047ad316a1c6c6fbd13ecf7a0abdcc375463920bd25bf5c142cc  ./oak_functions_freestanding_bin/target/x86_64-unknown-none/release/oak_functions_freestanding_bin
+```
+
+### Step 1: Download the binary
+
+The hash of the binary
+(`d059c38cea82047ad316a1c6c6fbd13ecf7a0abdcc375463920bd25bf5c142cc`) can be used
+to download the binary from `ent`, with the following command using the
+[ent HTTP API](https://github.com/google/ent#raw-http-api):
+
+```bash
+$ curl --output oak_functions_from_ent  https://ent-server-62sa4xcfia-ew.a.run.app/raw/sha256:d059c38cea82047ad316a1c6c6fbd13ecf7a0abdcc375463920bd25bf5c142cc
+  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                 Dload  Upload   Total   Spent    Left  Speed
+100 2327k    0 2327k    0     0  3637k      0 --:--:-- --:--:-- --:--:-- 3636k
+```
+
+### Step 2: Download the provenance
+
+The provenance can be downloaded from Rekor or the GitHub actions workflow that
```

Comment:
I wonder if we should consider also automatically uploading the provenance to ent at some point, so that we don't even have to interact with rekor at the time when we are trying to download the binary and its provenance. Anyways, let's discuss separately

Reply:
I think that is a good idea. We have to add a few steps to the GitHub workflow for this. Created #3339. I've also created #3333 to update the docs once the workflow is updated.